filters/canada/ontario/kitchener/public-art-geojson-to-json.js

```diff
@@ -1,40 +1,94 @@
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with url ${d.url}`);
+        let skip = false;
+        
+        // add name (required)
+        if (!add_required(item, "name", d, d.properties.MAP_LABEL, errors)) {
+            skip = true;
+        }
+
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, d.geometry.coordinates[0], errors)) {
+            skip = true;
         }
-        let coordinates = { longitude: d.geometry.coordinates[0], latitude: d.geometry.coordinates[1]};
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with url ${d.url}`);
+        if (!add_required(coordinates, "latitude", d, d.geometry.coordinates[1], errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
 
-        let details = {};
-        if (d.properties["Description"] != null || d.properties["Description"] != undefined) {
-            details.description = d.properties["Description"];
-        }
-        details.artist_first_name = d.properties["ARTIST_FIRST_NAME"];
-        details.artist_last_name = d.properties["ARTIST_LAST_NAME"];
-        details.location = d.properties["LOCATION"];
-        details.location_description = d.properties["LOCATION_DESCRIPTION"];
+        //optional fields
+        // item.artist = d.properties.ARTIST_FIRST_NAME + " " + d.properties.ARTIST_LAST_NAME
+        let artist_name;
+        if (d.properties.ARTIST_FIRST_NAME != null) artist_name = d.properties.ARTIST_FIRST_NAME;
+        if (d.properties.ARTIST_LAST_NAME != null) artist_name += " " + d.properties.ARTIST_LAST_NAME;
+        add_if_not_null(item, "artist", artist_name);
+
+
+        item.dates = create_dates_template();
 
-        item.details = details;
-        new_data.push(item);
+        add_if_not_null(item.dates.created, "year", d.properties["CONSTRUCTION_YEAR"]);
+        add_if_not_null(item.dates.installed, "year", d.properties["INSTALLATION_YEAR"]);
+
+        remove_null_date_fields(item);
+
+        add_if_not_null(item, "material", d.properties["MEDIUM"]);
+        add_if_not_null(item, "owner", d.properties["OWNERSHIP"]);
+        add_if_not_null(item, "area", d.properties["LOCATION"]);
+        add_if_not_null(item, "type", d.properties["TYPE"]);
+
+        // Not sure if "INITIAL_COST" refers to budget!
+        add_if_not_null(item, "budget", d.properties["INITIAL_COST"]);
+
+        item.address = {}
+        add_if_not_null(item.address, "street_address", d.properties["COMMON_LOCATION_REFERENCE"]);
+        add_if_not_null(item.address, "city", d.properties["MUNICIPALITY"]);
+        remove_if_empty(item, "address");
+
+        item.details = {}
+        add_if_not_null(item.details, "status", d.properties["STATUS"]);
+        remove_if_null(item.details);
```

Comment:
I think you meant to use the remove_if_empty function from the standard library here?